### PR TITLE
feat: add mvp bluesky tools

### PR DESCRIPTION
**Pull request by njhale**
Add Bluesky tools with support for basic search via unauthenticated requests.

Support for authenticated requests will require a custom OAuth provider that can handle [ATProto's authz requirements](https://atproto.com/specs/oauth#clients).

Support for datasets will be implemented as a follow up.

Depends on https://github.com/otto8-ai/otto8/pull/629

### CODE REVIEW

File: bluesky/tool.gpt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Param: query: A Lucene query to search for users with.
+Param: limit: (optional) The maximum number of users to return. Must be an integer >=1 and <=100. Defaults to 25.
+
+#!http://service.daemon.gptscript.local/tools/users/search
+
+---
+Name: service
+Metadata: index: false
+
+#!sys.daemon /usr/bin/env npm --prefix ${GPTSCRIPT_TOOL_DIR} run server
```
Comment (thedadams) [inline bluesky/tool.gpt:35]: Interesting that you chose to do this as a daemon. I don't think this works when we add authentication.

Comment (njhale) [reply bluesky/tool.gpt:35]: Oh, I had assumed the creds would get passed with the env header. I'll switch to a non-daemon

**Review by thedadams — APPROVED** [review]

File: bluesky/src/posts.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    if (limit !== undefined && limit !== '') {
+        try {
+            queryParams.limit = parseInt(limit, 10)
+        } catch (e: unknown) {
+            throw new Error(`Invalid limit format: ${String(e)}`)
+        }
+    }
+
+    if (until !== undefined && until !== '') {
```
[the remark this answers is not in the document]
Comment (njhale) [reply bluesky/src/posts.ts]: Done

File: bluesky/tool.gpt (hunk 1 above)
Comment (njhale) [reply bluesky/tool.gpt]: OK -- updated this to be a "command driven" tool

**Review by ryanhopperlowe — APPROVED** [review]

**Pull request merged**